### ISSUE TITLE
Two versions of the same resource map are both indexed in the `resourceMap` field for an aggregated object

### ISSUE THREAD

**Issue by laurenwalker** [NCEAS/metacat#1258]
This bug was first noticed by Val Hendrix who noticed that some packages have multiple download links in MetacatUI: https://github.com/NCEAS/metacatui/issues/537

## Bug description
When an EML document is aggregated in a resource map, it sometimes has two pids in its `resourceMap` field which are different versions of a resource map in the same obsolescence chain. (This may be happening to non-EML objects, too, but I've only noticed it on EML documents so far).

It's my understanding that the `resourceMap` Solr field should only contain the pid for the most recent resource map(s) that aggregates that object.

For example,

This EML document: [`https://dev.nceas.ucsb.edu/#view/urn:uuid:fb6f8236-e6f7-4957-a818-4d90dc3ce473`](https://dev.nceas.ucsb.edu/#view/urn:uuid:fb6f8236-e6f7-4957-a818-4d90dc3ce473) is only aggregated by one resource map. But if you look at the Solr document: [`https://dev.nceas.ucsb.edu/knb/d1/mn/v2/query/solr/?q=id:%22urn:uuid:fb6f8236-e6f7-4957-a818-4d90dc3ce473%22&fl=id,resourceMap`](https://dev.nceas.ucsb.edu/knb/d1/mn/v2/query/solr/?q=id:%22urn:uuid:fb6f8236-e6f7-4957-a818-4d90dc3ce473%22&fl=id,resourceMap) you will see that it has two values for the `resourceMap` field: [`urn:uuid:c89cf109-1b85-4b98-a4fd-1ce59196b49e`](https://dev.nceas.ucsb.edu/knb/d1/mn/v2/meta/urn:uuid:c89cf109-1b85-4b98-a4fd-1ce59196b49e) and [`urn:uuid:387956ba-f26a-4fdf-a69f-bb831bff0738`](https://dev.nceas.ucsb.edu/knb/d1/mn/v2/meta/urn:uuid:387956ba-f26a-4fdf-a69f-bb831bff0738).

```xml
<response>
<lst name="responseHeader">
<int name="status">0</int>
<int name="QTime">11</int>
<lst name="params">
<str name="fq">(readPermission:"public")OR(isPublic:true)</str>
<str name="fl">id,resourceMap</str>
<str name="q">id:"urn:uuid:fb6f8236-e6f7-4957-a818-4d90dc3ce473"</str>
</lst>
</lst>
<result name="response" numFound="1" start="0">
<doc>
<str name="id">urn:uuid:fb6f8236-e6f7-4957-a818-4d90dc3ce473</str>
<arr name="resourceMap">
<str>urn:uuid:c89cf109-1b85-4b98-a4fd-1ce59196b49e</str>
<str>urn:uuid:387956ba-f26a-4fdf-a69f-bb831bff0738</str>
</arr>
</doc>
</result>
</response>
```

`urn:uuid:c89cf109-1b85-4b98-a4fd-1ce59196b49e` is obsoleted by `urn:uuid:387956ba-f26a-4fdf-a69f-bb831bff0738`:

```xml
<ns3:systemMetadata xmlns:ns2="http://ns.dataone.org/service/types/v1" xmlns:ns3="http://ns.dataone.org/service/types/v2.0">
<serialVersion>1</serialVersion>
<identifier>urn:uuid:c89cf109-1b85-4b98-a4fd-1ce59196b49e</identifier>
<formatId>http://www.openarchives.org/ore/terms</formatId>
<size>7954</size>
<checksum algorithm="MD5">098fcf154731b5b37250efa4f4961aac</checksum>
<submitter>http://orcid.org/0000-0002-2192-403X</submitter>
<rightsHolder>http://orcid.org/0000-0002-2192-403X</rightsHolder>
<accessPolicy>
<allow>
<subject>public</subject>
<permission>read</permission>
</allow>
<allow>
<subject>http://orcid.org/0000-0003-2192-431X</subject>
<permission>read</permission>
<permission>write</permission>
<permission>changePermission</permission>
</allow>
</accessPolicy>
<replicationPolicy replicationAllowed="true" numberReplicas="3"/>
<obsoletes>
resource_map_urn:uuid:04a8c534-aa6b-4ff6-b21d-e96fb80b07ec
</obsoletes>
<obsoletedBy>urn:uuid:387956ba-f26a-4fdf-a69f-bb831bff0738</obsoletedBy>
<archived>false</archived>
<dateUploaded>2018-06-01T18:32:38.696+00:00</dateUploaded>
<dateSysMetadataModified>2018-06-01T18:33:16.615+00:00</dateSysMetadataModified>
<originMemberNode>urn:node:mnTestKNB</originMemberNode>
<authoritativeMemberNode>urn:node:mnTestKNB</authoritativeMemberNode>
<fileName>
resourceMap_urn:uuid:c89cf109-1b85-4b98-a4fd-1ce59196b49e.xml
</fileName>
</ns3:systemMetadata>
```

The only pid that should be indexed as the `resourceMap` for that EML document is `urn:uuid:387956ba-f26a-4fdf-a69f-bb831bff0738` since that is the most recent resource map that EML document is in.

I have a feeling it may happen when an EML and resource map are updated before one or both are fully indexed, but this is just a guess.

**Comment by taojing2002** [comment]
So we would NOT change the index behavior. But we will change MetacatUI:

Hi all -

Last week during our NCEAS dev meeting we talked about an issue with the download dropdown menus that appear in the search result rows in MetacatUI. The issue was originally brought to my attention by Val Hendrix with ESS-DIVE.  This issue is low-to-medium priority and effects DataONE, ESS-DIVE, KNB, and Arctic Data.

I typed up a description of this bug and some solutions here: https://hackmd.io/kcnViO5ZSEGx1u2L7aonBw?view

Feel free to share your input if you are interested. If I don't hear anything in the next 1-2 weeks, I am going to go ahead with solution C in the hackpad above.

Thanks,
Lauren
